### DISCUSSION TITLE
[Auth] Add search/filter by user id inside Auth table

### DISCUSSION THREAD

**Discussion by iosephmagno** [supabase/supabase#27536] [Feature Requests]
Due to bug https://github.com/supabase/auth/issues/1629, user failed to change his phone number. After several attempt using different phone numbers, the phone number eventually changed to a new one, but that was not the correct one. As consequence of that, the phone number inside table public.users is wrong for this user and we cannot understand what is the user's phone number registered to Auth table because Auth table cannot be filtered by user id. Adding search/filter by user id would help fix this case and would be helpful in general anyway We would appreciate if the feature could be added with upcoming releases.

<img width="1203" alt="Screenshot 2024-06-25 at 16 47 36" src="https://github.com/supabase/supabase/assets/83366073/45b7b1ee-b912-435e-8e63-182e7c12d558">

**Comment by GaryAustin1** [comment]
For now you can go the table UI.  Select auth schema, and the users table.  Then use the filters there to filter the phone column.

**Reply by iosephmagno** [reply]
Can you share a screenshot? I cannot find it

**Reply by iosephmagno** [reply]
I was able to fond the auth schema inside table UI bit the id in the table is not the user is, ao I cannot find which phone number the user is having now.

**Reply by GaryAustin1** [reply]
![image](https://github.com/supabase/supabase/assets/54564956/5f94075b-eb22-4287-8413-2d4f4aa6279f)
That is the user id auth uses.

**Reply by iosephmagno** [reply]
Then we have a serious issue. Not only that user is not in the table, but also other users from table public.users are not in the table 

<img width="927" alt="Screenshot 2024-06-25 at 18 29 21" src="https://github.com/supabase/supabase/assets/83366073/dbf51946-48cf-4869-952b-2160ab0a2cb8">

**Reply by iosephmagno** [reply]
Should we open a ticket and give you access to the db? This is a severe issue for us. It there is data inconsistency between public.users and auth table, then I cant image what will happen when user will try to delete his account and change his number. Indeed with opened bugs about that and things might be interwined.

**Reply by GaryAustin1** [reply]
What are you filtering on? It says one filter applied.

The authentication table you showed up top gets its information from the auth.users table so if it is in the authentication table it has to be in the auth.users table.

You can certainly generate a support request.  I am not an employee and have no access to your instance.

**Reply by iosephmagno** [reply]
I filter a user by userId. User is inside public.users table but is not inside auth. We think these issue occurs as consequence of bugs in "delete user" and "change user phone number". We filed this issues on github.

**Reply by GaryAustin1** [reply]
If you have a public user table then you have to make sure it gets cleaned up when you use auth.admin.deleteUser().  Supabase auth code does not know about your public users table.

**Reply by iosephmagno** [reply]
It is a foreign key cascade, if user gets deleted from auth, item gets deleted from public.users table.

Issue with deletion is that:
If user deletes himself and re-signup soon after with same number, supabase goes haywire

Also, if user changes his phone number and then change it back to previous one, supabase goes haywire. 

We opened two issues for that. It seems that this also causes the data inconstency between public.users and auth table. 

CC: @kiwicopple

**Reply by iosephmagno** [reply]
If our discovery is correct, this is pretty severe. Because the more users change their numbers and/or delete/signup back with same phone number, and the more `public.users` and `auth` tables will diverge (be inconsistent).

**Reply by iosephmagno** [reply]
https://github.com/supabase/auth/issues/1629

https://github.com/supabase/auth/issues/1627

**Reply by GaryAustin1** [reply]
My only comment on public.users is that you are responsible for all updates/deletions etc. on it.  If the user row is gone from the auth.users table then either you have to delete, or cascade delete has to take care of it if set up correctly.  If phone numbers change you have to have an update trigger to update public.users if you keep the phone numbers there.

I can't comment on your auth reports themselves.  I don't use phone numbers for users.

**Reply by GaryAustin1** [reply]
I'm not sure that your public.users table is not part of the issue.
You should look at the Postgres logs when you get the error.  
If you delete a user and their row stays in your public users table with the phone number there, then when you try to signup again you could get an error if the phone number column is marked unique.

You say there are rows in the public.users table with UUID's that are not in auth.users table.  That is not good for the typical public.users table and can lead to all kinds of failures if you check that any data has to be unique in that table.

**Reply by iosephmagno** [reply]
Are you sure that auth id is public.users id, coz you suggested that in a comment above but it doesnt seem so. In any events, our reported issues are correct.

**Reply by GaryAustin1** [reply]
I have no idea what is in public.users.  That is a table you created.  auth.users is Supabase's table that I showed and should match the authenticated tab table that highlights it.

**Comment by iosephmagno** [comment]
Please check above issues. Auth is buggy, if you dont signup user via otp codes you might not experience issues.
